invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/communities.js

```diff
@@ -50,67 +48,35 @@ function ResultsGridItemTemplate({ result, index }) {
 }
 
 export function CommunitiesResultsItemTemplate({ result, index }) {
-  const community_type = _get(
-    result,
-    "ui.type.title_l10n",
-    i18next.t("No community type")
-  );
+  const community_type = result.ui?.type?.title_l10n;
+  const visibility = result.access.visibility;
+  let visibilityColor = "red";
+  let visibilityText = i18next.t("Restricted");
+  let visibilityIcon = "ban";
+  if (visibility === "public") {
+    visibilityColor = "green";
+    visibilityText = i18next.t("Public");
+    visibilityIcon = undefined;
+  }
```

Comment:
I think this part needs a little simplification
```suggestion
const isPublic = visibility === "public";
const visibilityColor = isPublic ? "green": "red" ;
const visibilityText = isPublic ?  i18next.t("Public") : i18next.t("Restricted");
```
additionally I think you don't need the middleman component `CommunitiesResultsItemTemplate` you can directly use ComputerTabletCommunitiesItem and MobileCommunitiesItem inside the parent directly, and compute the values above in each component, making it more flexible and less complex